### PR TITLE
Added httptrace support

### PR DESCRIPTION
**Pull request by JRonak**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,4 +53,6 @@ Options:
   -cpus                 Number of used cpu cores.
                         (default for current machine is 8 cores)
   -host                 HTTP Host header.
+  -http-trace         	Enable http trace detailed info on various events during request (Experimental)
```
Comment (rakyll) [inline README.md]: httptrace is an terminology that makes more sense in the scope of the Go standard library.

Should we just rename this flag to reflect that it is providing just more info? A flag like -more?

File: requester/print.go [hunk 2]
```diff
@@ -32,8 +32,15 @@ type report struct {
...
 
+	avgConn  float64
+	avgDns   float64
+	avgReq   float64
+	avgRes   float64
+	avgDelay float64
+
 	results chan *result
 	total   time.Duration
+	trace   bool
```
Comment (rakyll) [inline requester/print.go]: move this right before avgConn bool and comment that if it is set, the following fields will be filled.

File: requester/requester.go [hunk 3]
```diff
@@ -83,36 +92,72 @@ func (b *Work) Run() {
...
 	var size int64
 	var code int
-
-	resp, err := c.Do(cloneRequest(b.Request, b.RequestBody))
+	var dnsDuration, connDuration, resDuration, reqDuration, delayTime time.Duration
+	req := cloneRequest(b.Request, b.RequestBody)
+	if b.EnableTrace {
+		trace := &httptrace.ClientTrace{
+			DNSStart: func(info httptrace.DNSStartInfo) {
+				dnsDuration = time.Now().Sub(s)
```
Comment (rakyll) [inline requester/requester.go]: reusing the same var for start is confuses the reader.

var dnsStart, connStart, ....
var dnsDuration, connDuration, resDuration, reqDuration, delayTime time.Duration

Comment (rakyll) [inline requester/requester.go]: dnsStart = time.Now()

File: requester/requester.go [hunk 4]
```diff
@@ -83,36 +92,72 @@ func (b *Work) Run() {
...
+			},
+			DNSDone: func(dnsInfo httptrace.DNSDoneInfo) {
+				t := time.Now().Sub(s)
+				dnsDuration = time.Duration(t.Nanoseconds() - dnsDuration.Nanoseconds())
```
Comment (rakyll) [inline requester/requester.go]: dnsDuration = time.Now().Sub(dnsStart)

File: README.md (hunk 1 above)
Comment (JRonak) [inline README.md]: I think more sounds fine. Suggest a tag?

File: requester/print.go (hunk 2 above)
Comment (JRonak) [inline requester/print.go]: I was thinking of making a separate struct type for  trace report, incase in future if we add extra fields like fastestDns, slowestDns etc. 
Rest of the mentioned changes above and below would be fixed soon

**Comment by JRonak** [comment]
Replaced http-trace with more tag and fixed the trace implementation.

**Comment by rakyll** [comment]
Is this your first time contributing to a Google project? If so, could you please review and sign the Google CLA? https://cla.developers.google.com/ to sign.

**Comment by JRonak** [comment]
Done, thanks

**Comment by rakyll** [comment]
Thanks!

**Pull request merged**